### PR TITLE
KO: add site/ko/guide/create_op.md (Korean translation)

### PR DESCRIPTION
**Pull request by wckim**
added guide doc, Korean translation of site/en/guide/create_op.md.

### CODE REVIEW

File: site/ko/guide/create_op.md [hunk 1]
```diff
@@ -0,0 +1,1347 @@
...
+           .Attr("T: numerictype = DT_FLOAT");
+
+2. 속성에 대한 제약을 줄여서 안전하게 만들 수 있습니다.
+   예를 들어, `{int32, int64}`을 `{int32, int64, float}` 혹은 `type`으로 변경가능합니다.
+   또는, `{"apple", "orange"}`을 `{"apple", "banana", "orange"}` 혹은 `string`으로 변경할 수 있습니다.
+
+3. 리스트의 기본값을 변경 이전과 동일하게 맞춘다면,
+   단일 입력/출력을 리스트로된 입력/출력으로 변경할 수 있습니다.
+
+4. 기본값이 없다면, 새로운 입력/출력 리스트룰 추가할 수 있습니다.
```
Comment (Choiuijin1125) [inline site/ko/guide/create_op.md]: "리스트룰" --> "리스트를" :)

File: site/ko/guide/create_op.md [hunk 2]
```diff
@@ -0,0 +1,1347 @@
...
+
+```bash
+TF_CFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_compile_flags()))') )
+TF_LFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_link_flags()))') )
+g++ -std=c++11 -shared zero_out.cc -o zero_out.so -fPIC ${TF_CFLAGS[@]} ${TF_LFLAGS[@]} -O2
+```
+
+Max OS X에서 `.so`을 생성할 때 "-undefined dynamic_lookup"이라는 추가적인 플래그가 필요합니다.
+
+>   `gcc` 버젼이 `>=5`인 경우 주의사항:
```
Comment (Choiuijin1125) [inline site/ko/guide/create_op.md]: "버젼" --> "버전" :)

File: site/ko/guide/create_op.md [hunk 3]
```diff
@@ -0,0 +1,1347 @@
...
+>    gcc는 버젼 `5`부터 새로운 C++ [ABI](https://gcc.gnu.org/gcc-5/changes.html#libstdcxx)를 사용합니다.
```
Comment (Choiuijin1125) [inline site/ko/guide/create_op.md]: "버젼" --> "버전" :)

File: site/ko/guide/create_op.md [hunk 4]
```diff
@@ -0,0 +1,1347 @@
...
+안전하거나 안전하지 않는 변경에 대한 전체 내용은
+[`tensorflow/core/framework/op_compatibility_test.cc`](https://www.tensorflow.org/code/tensorflow/core/framework/op_compatibility_test.cc)에서
+찾을 수 있습니다.
+만약에 하위 호환성을 확보하기 어렵다면, 그때는 새로운 이름과 의미를 갖는 새로운 연산을 만드세요.
+
+또한 이러한 변경을 `GraphDef` 호환성을 지키면서 유지할 수 있는 동안,
+생성되는 파이썬 코드는 이전 버전과 호환되지 않는 방식으로 변경될 수 있음을 주의하세요.
+새로운 선택형 파라미터를 추가하는 것을 제외하고 이전 버전의 특징을 유지함으로써,
+파이썬 API는 사용자가 파이썬 랩퍼를 신중하게 변경하여 호환성을 유지할 수 있습니다.
+일반적으로 호환되지 않는 변경은 텐서플로 메이저 버젼이 변경되고,
```
Comment (Choiuijin1125) [inline site/ko/guide/create_op.md]: "버젼" --> "버전" :)

File: site/ko/guide/create_op.md [hunk 5]
```diff
@@ -0,0 +1,1347 @@
...
+[`GraphDef` 버전 의미](./versions.md#compatibility_of_graphs_and_checkpoints)를 준수해야 하는 경우에만 발생합니다.
+
+### GPU 지원
+
+OpKernel을 구현할 때 CPU를 지원하는 것과 GPU를 지원하는 것으로 다르게 구현을 할 수 있고,
+이를 [다른 자료형을 위한 커널 등록](#다형성)과 같이 처리할 수 있습니다.
+[`tensorflow/core/kernels/`](https://www.tensorflow.org/code/tensorflow/core/kernels/)는
+GPU를 지원하는 커널의 예입니다.
+어떤 커널은 `.cc`로 끝나는 파일이 CPU버젼이고 `_gpu.cu.cc`로 끝나는 파일이 GPU버젼이고
```
Comment (Choiuijin1125) [inline site/ko/guide/create_op.md]: "버젼" --> "버전" :)

File: site/ko/guide/create_op.md [hunk 6]
```diff
@@ -0,0 +1,1347 @@
...
+
+예를 들어, `tf.pad`는 GPU 커널을 위한 부분을 제외하고 [`tensorflow/core/kernels/pad_op.cc`][pad_op]에 모든 구현되어 있습니다.
+[`tensorflow/core/kernels/pad_op_gpu.cu.cc`](https://www.tensorflow.org/code/tensorflow/core/kernels/pad_op_gpu.cu.cc)에
+GPU커널에 관련된 부분이 있고, 공유하는 코드는 템플릿 클래스 형태로
+[`tensorflow/core/kernels/pad_op.h`](https://www.tensorflow.org/code/tensorflow/core/kernels/pad_op.h)에 정의되어 있습니다.
+코드를 이처럼 관리하는 이유는 2가지입니다:
+CPU를 위한 구현과 GPU를 위한 구현에 필요한 공통 코드를 공유할 수 있고
+GPU 컴파일러가 컴파일하는 GPU구현 부분만 다른 파일로 분리할 수 있습니다.
+
+추가적으로 알아야할 것은, `pad`의 GPU 커널 버젼이 사용되는 경우에도,
```
Comment (Choiuijin1125) [inline site/ko/guide/create_op.md]: "버젼" --> "버전" :)

File: site/ko/guide/create_op.md [hunk 7]
```diff
@@ -0,0 +1,1347 @@
...
+
+```c++
+REGISTER_OP("ZeroOut")
+    .Input("to_zero: int32")
+    .Output("zeroed: int32")
+    .SetShapeFn(::tensorflow::shape_inference::UnchangedShape);
+```
+
+형태 함수는 입력을 형태를 제한할 수 있습니다.
+[`ZeroOut`의 입력을 벡터 형태로 제한한](#조건부-검사와-검증) 버젼의 경우, 형태 함수는 다음과 같을 것입니다:
```
Comment (Choiuijin1125) [inline site/ko/guide/create_op.md]: "버젼" --> "버전" :)

File: site/ko/guide/create_op.md [hunk 8]
```diff
@@ -0,0 +1,1347 @@
...
+만약에 쿠다 라이브러리가 `/usr/local/lib64`에 설치되어 있지 않다면,
+위의 2번째 명령어(g++)에 라이브러리 경로를 명시적으로 추가할 수 있습니다.
+예를 들어, 쿠다가 `/usr/local/cuda-8.0`에 설치되어 있다면 `-L /usr/local/cuda-8.0/lib64/`를 추가하세요.
+
+>   일부 리눅스에서는, `nvcc` 컴파일 단계에서 추가적인 설정이 필요합니다.
+>   `nvcc` 명령어에서 `mwaitxintrin.h`때문에 생길 수 있는 에러를 피하기 위해 `-D_MWAITXINTRIN_H_INCLUDED`를 추가하세요.
+
+### 파이썬에서 그래디언트 구현하기
+
+연산 그래프가 주어지면, 텐서플로는 기존 연산에 대한 그래디언트를 표현하는 새로운 연산을 추가하기 위해서 자동미분(역전파) 사용합니다.
```
Comment (Choiuijin1125) [inline site/ko/guide/create_op.md]: "자동미분(역전파)" --> "자동미분(역전파)을"

File: site/ko/guide/create_op.md [hunk 9]
```diff
@@ -0,0 +1,1347 @@
...
+  예를 들어, 실수형 텐서 `x`와 정수 인덱스 `i`를 입력으로 받아들이는 연산인 경우
+  그래디언트 함수는 `return [x_grad, None]`일 것입니다.
+
+* 연산을 위해 의미가 있는 그래디언트가 없다면, 보통 어떤 그래디언트도 등록하지 않을 것이고,
+  연산의 그래디언트가 전혀 필요없는 한 그 것은 괜찮을 것입니다.
+  어떤 경우에, 연산은 잘 정의된 그래디언트를 가지고 있지 않지만 그래디언트의 계산에 관여될 수 있습니다.
+  그런 경우 자동적으로 0을 역전파하기 위해서 `ops.NotDifferentiable`를 사용할 수 있습니다.
+
+그 때 그래디언트 함수가 호출 될 때 텐서 그 자체가 아니라 연산 데이터 흐름 그래프만 사용할 수 있음을 유의하세요.
+그래서, 모든 계산은 그래프 실행시간에 실행되기 위해서 다른 텐스플로 연산을 사용해서 수행해야 합니다.
```
Comment (Choiuijin1125) [inline site/ko/guide/create_op.md]: "텐스플로" --> "텐서플로" :)

File: site/ko/guide/create_op.md [hunk 10]
```diff
@@ -0,0 +1,1347 @@
...
+다음과 같이 사용됩니다:
+
+```c++
+REGISTER_OP("ZeroOut")
+    .Input("to_zero: int32")
+    .Output("zeroed: int32")
+    .SetShapeFn(::tensorflow::shape_inference::UnchangedShape);
+```
+
+형태 함수는 입력을 형태를 제한할 수 있습니다.
```
Comment (Choiuijin1125) [inline site/ko/guide/create_op.md]: "입력을 형태를" --> "입력의 형태를" :)

File: site/ko/guide/create_op.md [hunk 11]
```diff
@@ -0,0 +1,1347 @@
...
+.SetShapeFn([](::tensorflow::shape_inference::InferenceContext* c) {
+    c->set_output(0, c->Matrix(c->Dim(c->input(0), 0), 3));
+    return Status::OK();
+});
+```
+
+만약 복잡한 형태 함수가 있다면,
+다양한 입력 형태 조합이 기대한 출력 형태 조합을 만들어 내는지 검증하기 위한 테스트를 추가하는 것을 고려해야 합니다.
+그러한 테스트를 어떻게 작성하는지에 대한 예는
+[core ops tests](https://www.tensorflow.org/code/tensorflow/core/ops/array_ops_test.cc)에서 볼 수 있습니다
```
Comment (Choiuijin1125) [inline site/ko/guide/create_op.md]: I think it's better to translate "[core ops tests]" :0 --> "[주요 연산 테스트]"

**Comment by lamberta** [comment]
Thank you.

Community translations are now located in the [tensorflow/docs-l10n](https://github.com/tensorflow/docs-l10n) repo (see [this message](https://groups.google.com/a/tensorflow.org/g/docs/c/GJPq7S_fw4c/m/lA1qfBX2EQAJ) for context).

I'll move over/duplicate existing pull requests as I find time. In the meantime, please feel free to create a new pull request in that repo 😀 Then you can close this PR and add a link to the new one.

**Comment by lamberta** [comment]
Migrated in https://github.com/tensorflow/docs-l10n/pull/87

**Pull request closed without merging**